src/Fmt_ast.ml

```diff
@@ -2269,11 +2269,14 @@ and fmt_cases c ctx cs =
             (fmt_or_k paren_body (fmt "@;<1 2>->") (fmt " ->@;<0 3>"))
             (fmt_or_k paren_body (fmt "@;<1 -2>-> (") (fmt " ->@;<0 -1>"))
         in
+        let pro =
+          fmt_or_k c.conf.break_cases
+            (break_unless_newline 100 0 $ fmt "| ")
```

Comment:
Elsewhere in the code, 1000 is used to force-break a box. It may be that 100 is close enough to a width someone will actually use that it could fit and then look very bad.